store.go

```diff
@@ -3417,12 +3417,44 @@ func ReloadConfigurationFile(configFile string, storeOptions *StoreOptions) {
 	}
 }
 
+var (
+	prevStoreOptions *StoreOptions
+	prevMod          time.Time
+	prevMutex        sync.Mutex
+	prevConfigFile   string
+)
+
+func maybeReloadConfigurationFile(configFile string, storeOptions *StoreOptions) {
+	prevMutex.Lock()
+	defer prevMutex.Unlock()
+
+	fi, err := os.Stat(configFile)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			fmt.Printf("Failed to read %s %v\n", configFile, err.Error())
+		}
+		return
+	}
+
+	mtime := fi.ModTime()
+	if prevStoreOptions != nil && prevMod == mtime {
```

Comment:
Shouldn't you test if prevConfigFIle == configFile?

Reply:
good catch.  Yes I need to do that.  Pushed a fixed version